arch/arm64/core/Kconfig

```diff
@@ -101,6 +115,29 @@ config ARMV8_A
 	  so that it can support some features included in the AArch64 state.
 	  It supports the T32 and A32 instruction sets.
 
+endif # CPU_CORTEX_A
+
+if CPU_CORTEX_R_AARCH64
+
+config ARMV8_R
+	bool
+	select ATOMIC_OPERATIONS_BUILTIN
+	select CPU_HAS_MPU
```

Comment:
Is this a peripheral always present in the SOC implementations of this arhitecture?

Reply:
Thanks for your kind reminder!
1. For the ATOMIC part, Armv8r is exactly the same as Armv8a, so following the Armv8a, I think ATOMIC_OPERATIONS_BUILTIN should be here?
2. But for the MPU part, `CPU_HAS_MPU` should not. Since MPU is an optional config for cortex-R82, I plan to move the `select CPU_HAS_MPU` into soc relevant Kconfigs?
May I ask your suggestions?

Reply:
> * For the ATOMIC part, Armv8r is exactly the same as Armv8a, so following the Armv8a, I think ATOMIC_OPERATIONS_BUILTIN should be here?

That's fine.

> * But for the MPU part, `CPU_HAS_MPU` should not. Since MPU is an optional config for cortex-R82, I plan to move the `select CPU_HAS_MPU` into soc relevant Kconfigs?
>   May I ask your suggestions?

Yes, moving it to the SOC Kconfig is a good idea.